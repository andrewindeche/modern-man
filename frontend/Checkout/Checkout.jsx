import React from 'react';
import NotificationBar from '../src/components/NotificationBar';
import SearchBar from '../src/components/SearchBar';
import NavButtons from '../src/components/NavButtons';
import Mpesa from '../src/images/mastercard.png';
import Visa from '../src/images/mpesa.png';
import Mastercard from '../src/images/paypal.png';
import Paypal from '../src/images/visa.png';
import Suit1 from '../src/images/tuxedo1.jpg';
import Suit3 from '../src/images/white tux.jpg';

const Checkout = () => {
    return(
        <>
          <NotificationBar />
          <SearchBar />
          <NavButtons />
          <div className='checkout'>
          <h1 id="Checkout">Check Out</h1>
          <div className='personaldetails'>
          <h4 className='checkouttitle'>Personal Information</h4>
            <ul className='aboutuser'>
            <p className='about'>About</p>
              <li>Andrew Indeche</li>
              <li>Westlands</li>
              <li>Nairobi, Kenya </li>
              <li id='email'>Indecheandrew@gmail.com</li>
            </ul>
          <div className='payments'>
          <p>Select Payment</p>
          <img src={Mpesa} alt="Mpesa" />
          <img src={Visa} alt="Visa" />
          <img src={Paypal} alt="Paypal" />
          <img src={Mastercard} alt="Mastercard" />
          </div>
          <div className='paymentselected'>
          <h4 className='checkouttitle'>Payment Selected</h4>
          <img src={Mpesa} alt="Blue Tuxedo" />
            </div>
          </div>
          <div className="cartdetails">
            <h4>Shopping Cart Details</h4>
            <span className='checkoutitem'><img src={Suit1} alt="tuxedo"/>
            <p>Black Full Italian Men's Tuxedo * 2 </p>
            <a>$2000</a></span>
            <span className='checkoutitem'><img src={Suit1} alt="tuxedo" />
            <p>Grey full Official Men's Suit * 1</p>
            <a>$1500</a></span>
            <span className='checkoutitem'><img src={Suit3} alt="tuxedo" />
            <p>White full Official Men's Suit* 1</p>
            <a>$1800</a></span>
            </div>
            <div className='grandtotal'>
              <span className='checkoutamount'><p>SubTotal</p>
              <a>$3400</a></span>
              <span className='checkoutamount'><p>Shipping</p>
              <a>$0</a></span>
              <span className='checkoutamount'><p>Total</p>
              <a>$3400</a></span>
            </div>
        </div>
        </>
    )
}
export default Checkout