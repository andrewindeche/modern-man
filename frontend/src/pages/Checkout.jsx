import NotificationBar from '../components/NotificationBar';
import SearchBar from '../components/SearchBar';
import NavButtons from '../components/NavButtons';
import Mpesa from '../images/mastercard.webp';
import Visa from '../images/mpesa.webp';
import Mastercard from '../images/paypal.webp';
import Paypal from '../images/visa.webp';
import React from 'react';
import {
   faShoppingCart
} from '@fortawesome/free-solid-svg-icons';
const Checkout = () => {
  return(
    <>
      <NotificationBar />
      <SearchBar />
      <NavButtons />
      <div className='checkout'>
        <div className='checkout-container'>
        <div className='personaldetails'>
        <h4 className='checkouttitle'>User Profile</h4>
          <ul className='aboutuser'>
          <p className='about'>About</p>
            <li>Andrew Indeche</li>
            <li>Westlands</li>
            <li>Nairobi, Kenya </li>
            <li id='email'>Indecheandrew@gmail.com</li>
            </ul>
        <div className='payments'>
          <p>Select Payment</p>
          <img src={Mpesa} alt="Mpesa" />
          <img src={Visa} alt="Visa" />
          <img src={Paypal} alt="Paypal" />
          <img src={Mastercard} alt="Mastercard" />
            </div>
        <div className='paymentselected'>
          <h4 className='checkouttitle'>Payment Selected</h4>
          <img src={Mpesa} alt="Blue Tuxedo" />
            </div>
        </div>
        <div className='checkoutdetails'>
          <div className="cartdetails">
            <h4>Cart Summary</h4>
            <span className='checkoutitem'><img alt="tuxedo"/>
            <p>Black Full Italian Men's Tuxedo * 2 </p>
            <a>$2000</a></span>
            <span className='checkoutitem'><img alt="tuxedo" />
            <p>Grey full Official Men's Suit * 1</p>
            <a>$1500</a></span>
            <span className='checkoutitem'><img alt="tuxedo" />
            <p>White full Official Men's Suit* 1</p>
            <a>$1800</a></span>
            </div>
            <div className='grandtotal'>
              <span className='checkoutamount'><p>SubTotal</p>
              <a>$3400</a></span>
              <span className='checkoutamount'><p>Shipping</p>
              <a>$0</a></span>
              <span className='checkoutamount'><p>Total</p>
              <a>$3400</a>
              </span>
              <span className='checkoutamount'><button>Place Order</button></span>
            </div>
          </div>
        </div>
      </div>
        </>
    )
}
export default Checkout
