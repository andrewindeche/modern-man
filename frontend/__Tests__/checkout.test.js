import React from 'react';
import Checkout from 'pages/Checkout.jsx'; 
import renderer from 'react-test-renderer';

describe("checkout page",() =>{
  it('renders the checkout page', () => {
    const tree = renderer.create(<Checkout />).toJSON();
    expect(tree).toMatchSnapshot();
  })
})