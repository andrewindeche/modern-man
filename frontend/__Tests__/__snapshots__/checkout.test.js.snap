// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkout page renders the checkout page 1`] = `
[
  <div
    className="notificationbar"
  >
    <img
      alt="My Shoes"
      className="shoes"
      src=""
    />
    <p>
      Get 50% off Selected Shoes
    </p>
    <button
      id="notificationbutton"
    >
      Learn More
    </button>
  </div>,
  <div
    className="searchbar"
  >
    <h3>
      Modern Man
    </h3>
    <form
      className="search-form"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-magnifying-glass search-icon"
        data-icon="magnifying-glass"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={{}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      <input
        className="search-input"
        placeholder="Search for Men's Wear and Accessories"
        type="text"
      />
    </form>
    <div
      className="user-icons"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-user user"
        data-icon="user"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={{}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224 256A128 128 0 1 0 224 0a128 128 0 1 0 0 256zm-45.7 48C79.8 304 0 383.8 0 482.3C0 498.7 13.3 512 29.7 512H418.3c16.4 0 29.7-13.3 29.7-29.7C448 383.8 368.2 304 269.7 304H178.3z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      <span
        className="tooltip-text"
      >
        User Profile
      </span>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-cart-shopping shopping"
        data-icon="cart-shopping"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={{}}
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 24C0 10.7 10.7 0 24 0H69.5c22 0 41.5 12.8 50.6 32h411c26.3 0 45.5 25 38.6 50.4l-41 152.3c-8.5 31.4-37 53.3-69.5 53.3H170.7l5.4 28.5c2.2 11.3 12.1 19.5 23.6 19.5H488c13.3 0 24 10.7 24 24s-10.7 24-24 24H199.7c-34.6 0-64.3-24.6-70.7-58.5L77.4 54.5c-.7-3.8-4-6.5-7.9-6.5H24C10.7 48 0 37.3 0 24zM128 464a48 48 0 1 1 96 0 48 48 0 1 1 -96 0zm336-48a48 48 0 1 1 0 96 48 48 0 1 1 0-96z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      <span
        className="tooltip-text"
      >
        Checkout
      </span>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-heart heart"
        data-icon="heart"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={{}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M47.6 300.4L228.3 469.1c7.5 7 17.4 10.9 27.7 10.9s20.2-3.9 27.7-10.9L464.4 300.4c30.4-28.3 47.6-68 47.6-109.5v-5.8c0-69.9-50.5-129.5-119.4-141C347 36.5 300.6 51.4 268 84L256 96 244 84c-32.6-32.6-79-47.5-124.6-39.9C50.5 55.6 0 115.2 0 185.1v5.8c0 41.5 17.2 81.2 47.6 109.5z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      <span
        className="tooltip-text"
      >
        Favorite
      </span>
    </div>
  </div>,
  <div
    className="categories"
  >
    <button
      className="suits-image"
      style={
        {
          "backgroundImage": "url()",
        }
      }
    >
      Suits
    </button>
    <button
      className="shirts-image"
      style={
        {
          "backgroundImage": "url()",
        }
      }
    >
      Shirts
    </button>
    <button
      className="neck-wear"
      style={
        {
          "backgroundImage": "url()",
        }
      }
    >
      Neck wear & Accessories
    </button>
    <button
      className="shoes-image"
      style={
        {
          "backgroundImage": "url()",
        }
      }
    >
      Shoes
    </button>
  </div>,
  <div
    className="checkout"
  >
    <h1
      id="Checkout"
    >
      Check Out
    </h1>
    <div
      className="checkout-container"
    >
      <div
        className="personaldetails"
      >
        <h4
          className="checkouttitle"
        >
          Personal Information
        </h4>
        <ul
          className="aboutuser"
        >
          <p
            className="about"
          >
            About
          </p>
          <li>
            Andrew Indeche
          </li>
          <li>
            Westlands
          </li>
          <li>
            Nairobi, Kenya 
          </li>
          <li
            id="email"
          >
            Indecheandrew@gmail.com
          </li>
        </ul>
        <div
          className="payments"
        >
          <p>
            Select Payment
          </p>
          <img
            alt="Mpesa"
            src=""
          />
          <img
            alt="Visa"
            src=""
          />
          <img
            alt="Paypal"
            src=""
          />
          <img
            alt="Mastercard"
            src=""
          />
        </div>
        <div
          className="paymentselected"
        >
          <h4
            className="checkouttitle"
          >
            Payment Selected
          </h4>
          <img
            alt="Blue Tuxedo"
            src=""
          />
        </div>
      </div>
      <div
        className="checkoutdetails"
      >
        <div
          className="cartdetails"
        >
          <h4>
            Shopping Cart Details
          </h4>
          <span
            className="checkoutitem"
          >
            <img
              alt="tuxedo"
            />
            <p>
              Black Full Italian Men's Tuxedo * 2 
            </p>
            <a>
              $2000
            </a>
          </span>
          <span
            className="checkoutitem"
          >
            <img
              alt="tuxedo"
            />
            <p>
              Grey full Official Men's Suit * 1
            </p>
            <a>
              $1500
            </a>
          </span>
          <span
            className="checkoutitem"
          >
            <img
              alt="tuxedo"
            />
            <p>
              White full Official Men's Suit* 1
            </p>
            <a>
              $1800
            </a>
          </span>
        </div>
        <div
          className="grandtotal"
        >
          <span
            className="checkoutamount"
          >
            <p>
              SubTotal
            </p>
            <a>
              $3400
            </a>
          </span>
          <span
            className="checkoutamount"
          >
            <p>
              Shipping
            </p>
            <a>
              $0
            </a>
          </span>
          <span
            className="checkoutamount"
          >
            <p>
              Total
            </p>
            <a>
              $3400
            </a>
          </span>
          <span
            className="checkoutamount"
          >
            <button>
              Place Order
            </button>
          </span>
        </div>
      </div>
    </div>
  </div>,
]
`;
